BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+package com.braintreepayments.api
+
+import androidx.annotation.RestrictTo
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Contains the remote configuration for the Braintree Android SDK.
+ */
+// NEXT MAJOR VERSION: remove 'open' modifiers, Java classes/methods are open by default
+// Ref: https://kotlin-quick-reference.com/102c-R-open-final-classes.html
+open class Configuration internal constructor(configurationString: String?) {
```

Comment:
General question, can the `configurationString` be a required string instead of optional?

Reply:
In the future yeah, but in the `ConfigurationLoader` class it is possible that `null` gets passed in. It broke one of the tests so for now decided to keep it optional.

Reply:
Gotcha, thanks for the explanation - do we want to add a next major version for that as well? I know for iOS it's been nice to have a list of some todo's that we want to revisit in a breaking change.

Reply:
Def yeah. Actually I'll add a `TODO` to fix it when we migrate `ConfigurationLoader` over to Kotlin. This constructor is internal so it won't break anything.